e2e/ts_library/source-maps/source_map_check.ts

```diff
@@ -0,0 +1,39 @@
+import * as fs from 'fs';
+import * as path from 'path';
+const sourceMapResolve = require('source-map-resolve');
+
+const DEBUG = false;
+
+/**
+ * Checks if the source map for this file points to it's orignal source file
+ */
+function main() {
+  const workspaceDir = process.env.BUILD_WORKSPACE_DIRECTORY || '';
+
+  const runFileContent = fs.readFileSync(__filename).toString();
+  const maps = sourceMapResolve.resolveSourceMapSync(runFileContent, __filename, fs.readFileSync);
+
+  if (maps.map.sources.length !== 1) {
+    throw new Error('expected only one source file')
+  }
+  const sourceMapRelativePath = maps.map.sources[0];
+  const sourceMapRoot = maps.map.sourceRoot;
+  const fullSourcePath = path.join(sourceMapRoot, sourceMapRelativePath);
```

Comment:
for reference this is what im using as a reference as to what vscode I think
https://github.com/microsoft/vscode/blob/3fcb671444b82668ca64a15a7a21b1641170cf50/build/lib/nls.js#L66